utils/paramfetch/src/lib.rs

```diff
@@ -147,19 +147,34 @@ async fn fetch_verify_params(
 async fn fetch_params(path: &Path, info: &ParameterData) -> Result<(), anyhow::Error> {
     let gw = std::env::var(GATEWAY_ENV).unwrap_or_else(|_| GATEWAY.to_owned());
     debug!("Fetching {:?} from {}", path, gw);
-
-    let file = File::create(path).await?;
-    let mut writer = BufWriter::new(file);
-
     let url = format!("{}{}", gw, info.cid);
 
-    let client: surf::Client = surf::Config::default().set_timeout(None).try_into()?;
-
-    let req = client.get(url.as_str());
-
-    let mut source = req.await.map_err(|e| anyhow::anyhow!(e))?.compat();
-    copy(&mut source, &mut writer).await?;
+    let mut r = Ok(());
+    for _ in 0..5 {
```

Comment:
can be overkill, but can try using https://github.com/ihrwein/backoff

Reply:
Fixed